lib/src/metta/runner/stdlib/stdlib.metta

```diff
@@ -667,6 +667,34 @@
    (let $c1 (collapse $arg1) (let $c2 (collapse $arg2)
      (let $u (subtraction-atom $c1 $c2) (superpose $u)))))
 
+(@doc assert
+  (@desc "Asserts that the result of evaluating an expression is True")
+  (@params (
+    (@param "Expression to evaluate")))
+  (@return "Unit atom if the expression evaluates to True, error - otherwise"))
+(: assert (-> Expression Atom))
```

Comment:
Note that with that type signature, the following
```scheme
!(assert True)
```
might fail, because `True` is not an expression (an expression is basically a tuple), thus you probably want
```scheme
(: assert (-> Atom Atom))
```
or maybe
```scheme
(: assert (-> Atom (->))
```
where `(->)` seems to denote the unit type (I'm not completely sure about that though, maybe @vsbogd could shed some light).